application/src/Site/Navigation/Page/UriWithQuery.php

```diff
@@ -0,0 +1,36 @@
+<?php
+namespace Omeka\Site\Navigation\Page;
+
+use Laminas\Http\Request;
+use Laminas\Navigation\Page\AbstractPage;
+use Laminas\Navigation\Page\Uri;
+use Laminas\Uri\UriFactory;
+
+class UriWithQuery extends Uri
+{
+    public function isActive($recursive = false)
+    {
+        if (!$this->active) {
+            if ($this->getRequest() instanceof Request) {
+                $uriCurrent = $this->getRequest()->getUri();
+                $uriCurrentQuery = $uriCurrent->getQueryAsArray();
+                unset($uriCurrentQuery['page'], $uriCurrentQuery['sort_by'], $uriCurrentQuery['sort_order']);
+                $uriCurrent->setQuery($uriCurrentQuery);
+
+                $uriPage = UriFactory::factory($this->getUri());
+                $uriPageQuery = $uriPage->getQueryAsArray();
+                unset($uriPageQuery['page'], $uriPageQuery['sort_by'], $uriPageQuery['sort_order']);
+                $uriPage->setQuery($uriPageQuery);
+
+                $identicalPaths = $uriCurrent->getPath() === $uriPage->getPath();
```

Comment:
Slight concern about this overmatching... it wouldn't come up in the context its being used here for just browse pages, but the original Uri compares the request path to the full uri set to the nav page, which accounts for making things like absolute uris never "active" as they'd never match. Here if an absolute URI uses this class it would be "active" as long as the paths match even if it has a scheme/host/etc.